depthai_nodes/ml/parsers/utils/yolo.py

```diff
@@ -173,27 +188,46 @@ def non_max_suppression(
 def parse_yolo_outputs(
     outputs: List[np.ndarray],
     strides: List[int],
-    anchors: np.ndarray,
+    no: int,
+    anchors: Optional[List[np.ndarray]] = None,
     kpts: Optional[List[np.ndarray]] = None,
+    det_mode: bool = False,
+    yolo_version: YOLOVersion = YOLOVersion.DEFAULT,
 ) -> np.ndarray:
     """Parse all outputs of an YOLO model (all channels).
 
     @param outputs: List of outputs of an YOLO model.
     @type outputs: List[np.ndarray]
     @param strides: List of strides.
     @type strides: List[int]
-    @param anchors: List of anchors.
-    @type anchors: np.ndarray
+    @param no: Number of outputs of the model.
+    @type no: int
+    @param anchors: An optional list of anchors.
+    @type anchors: Optional[List[np.ndarray]]
     @param kpts: An optional list of keypoints for each output.
     @type kpts: Optional[List[np.ndarray]]
+    @param det_mode: Detection only mode.
+    @type det_mode: bool
+    @param yolo_version: YOLO version.
+    @type yolo_version: YOLOVersion
     @return: Parsed output.
     @rtype: np.ndarray
     """
     output = None
 
-    for i, (x, s, a) in enumerate(zip(outputs, strides, anchors)):
-        kpt = kpts[i] if kpts is not None else None
-        out = parse_yolo_output(x, s, a, head_id=i, kpts=kpt)
+    for i, (x, s) in enumerate(zip(outputs, strides)):
+        kpt = kpts[i] if kpts else None
+        a = anchors[i] if anchors else None
+        out = parse_yolo_output(
+            x,
+            s,
+            no,
+            a,
```

Comment:
personally, I'd make the variable names more meaningful. One has to invest time into figuring out what `x`, `s`, `no`, `a` means which is not optimal.